src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -511,7 +512,12 @@ private Map<Integer, Float> doANNSearch(
         FilterIdsSelector.FilterIdsSelectorType filterType = filterIdsSelector.getFilterType();
         // Now that we have the allocation, we need to readLock it
         indexAllocation.readLock();
-        indexAllocation.incRef();
+        try {
+            indexAllocation.incRef();
+        } catch (IllegalStateException e) {
```

Comment:
also we should log the exception as an error.

Reply:
makes sense to me.

Reply:
LGTM